tests/certification/pubsub/azure/servicebus/servicebus_test.go

```diff
@@ -0,0 +1,1136 @@
+		Step("verify if app1 has recevied messages published to newly created topic", assertMessages(10*time.Second, consumerGroup1)).
+		Run()
+}
+
+func TestServicebusEntityManagement(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+	consumerGroup1 := watcher.NewUnordered()
+
+	// Set the partition key on all messages so they are written to the same partition. This allows for checking of ordered messages.
+	metadata := map[string]string{
+		messageKey: partition0,
+	}
+
+	publishMessages := func(metadata map[string]string, sidecarName string, topicName string, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// prepare the messages
+			messages := make([]string, numMessages)
+			for i := range messages {
+				messages[i] = fmt.Sprintf("partitionKey: %s, message for topic: %s, index: %03d, uniqueId: %s", metadata[messageKey], topicName, i, uuid.New().String())
+			}
+
+			// add the messages as expectations to the watchers
+			for _, messageWatcher := range messageWatchers {
+				messageWatcher.ExpectStrings(messages...)
+			}
+
+			// get the sidecar (dapr) client
+			client := sidecar.GetClient(ctx, sidecarName)
+
+			// publish messages
+			ctx.Logf("Publishing messages. sidecarName: %s, topicName: %s", sidecarName, topicName)
+
+			var publishOptions dapr.PublishEventOption
+
+			if metadata != nil {
+				publishOptions = dapr.PublishEventWithMetadata(metadata)
+			}
+
+			for _, message := range messages {
+				ctx.Logf("Publishing: %q", message)
+				var err error
+
+				if publishOptions != nil {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message, publishOptions)
+				} else {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message)
+				}
+				require.Error(ctx, err, "error publishing message")
+			}
+			return nil
+		}
+	}
+
+	testDefault := func(ctx flow.Context) error {
+		ports, _ := dapr_testing.GetFreePorts(6)
+		grpcPortS1 := ports[0]
+		httpPortS1 := ports[1]
+		profilePortS1 := ports[2]
+		sidecar := sidecar.New(sidecarName1, embedded.WithComponentsPath("./components/entity_mgnt"),
+			embedded.WithoutApp(),
+			embedded.WithDaprGRPCPort(grpcPortS1),
+			embedded.WithDaprHTTPPort(httpPortS1),
+			embedded.WithProfilePort(profilePortS1),
+			runtime.WithPubSubs(
+				pubsub_loader.New("azure.servicebus", func() pubsub.PubSub {
+					return pubsub_servicebus.NewAzureServiceBus(log)
+				}),
+			),
+			runtime.WithSecretStores(secretstores_loader.New("local.env", func() secretstores.SecretStore {
+				return secretstore_env.NewEnvSecretStore(log)
+			})))
+		sidecar.Start(ctx)
+		return nil
+	}
+
+	flow.New(t, "servicebus certification - entity management enabled").
+
+		// Test : disableEntityManagement optional attribute , Test processing with all optional attributes and single publisher/subscriber
+		// Run subscriberApplication app1
+		Step("verify if app1 has recevied messages published to newly created topic", testDefault).
```

Comment:
Why are you not starting the sidecar directly in the flow like everywhere else? The sidecar should be initiated normally before you try to send an event

Reply:
As you explained this was done in a separate function because the sidecar cannot start up. If the sidecar cannot start (as expected) we should try to get the sidecar init failure and assert that instead. No point in attempting to publish messages which of course will fail and therefore that assert is somewhat meaningless. Any number of things could lead a sidecar to fail to startup and this test wouldn't be that meaningful.